src/onelogin/saml2/settings.py

```diff
@@ -214,26 +232,27 @@ def get_schemas_path(self):
 
     def _load_settings_from_dict(self, settings):
         """
-        Loads settings info from a settings Dict
+        Loads settings info from a settings Dict, adds default values and validates the settings
 
         :param settings: SAML Toolkit Settings
         :type settings: dict
 
         :returns: True if the settings info is valid
         :rtype: boolean
         """
+        self._sp = settings.get('sp', {})
```

Comment:
Order should not be changed.

Reply:
The order doesn't matter to the functionality, as the '_load_settings_from_dict' method is only called during initialization. To make the code easier I use the `get_idp_cert()` method to check if the config option or file actually exists and has content. Using the old order we would require way more code as the `get_idp_cert()` uses the data from the `_idp` dict (and so does the `get_sp_cert()` method)